qfieldsync/gui/layers_config_widget.py

```diff
@@ -248,10 +252,19 @@ def toggleMenu_triggered(self, action):
 
     def _on_dirtyset_wrapper(self):
         def _on_dirtyset():
-            for layer_source in self.layer_sources:
-                layer_source.read_layer()
-
-            self.reloadProject()
+            # the layer might got deleted by the time dirtyset is called
+            try:
+                for layer_source in self.layer_sources:
+                    layer_source.read_layer()
+
+                self.reloadProject()
+            except Exception:
```

Comment:
Can we use a more specific exception here?

Reply:
`RuntimeError`, better than nothing